sklearn/linear_model/_logistic.py

```diff
@@ -746,19 +750,32 @@ def grad(x, *args): return _multinomial_loss_grad(x, *args)[1]
             func = _logistic_loss
             def grad(x, *args): return _logistic_loss_and_grad(x, *args)[1]
             hess = _logistic_grad_hess
+        elif solver == 'trust-ncg':
+            func = _logistic_loss_and_grad
+            hess = _logistic_grad_hess
         warm_start_sag = {'coef': np.expand_dims(w0, axis=1)}
 
     coefs = list()
     n_iter = np.zeros(len(Cs), dtype=np.int32)
     for i, C in enumerate(Cs):
-        if solver == 'lbfgs':
+        if solver in ['lbfgs', 'trust-ncg']:
             iprint = [-1, 50, 1, 100, 101][
                 np.searchsorted(np.array([0, 1, 2, 3]), verbose)]
-            opt_res = optimize.minimize(
-                func, w0, method="L-BFGS-B", jac=True,
-                args=(X, target, 1. / C, sample_weight),
-                options={"iprint": iprint, "gtol": tol, "maxiter": max_iter}
-            )
+            if solver == 'lbfgs':
+                opt_res = optimize.minimize(
+                    func, w0, method="L-BFGS-B", jac=True,
+                    args=(X, target, 1. / C, sample_weight),
+                    options={"iprint": iprint, "gtol": tol,
+                             "maxiter": max_iter}
+                )
+            elif solver == 'trust-ncg':
+                def hessp(*args):
```

Comment:
When looking at the [scipy docs for minimize](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.minimize.html?highlight=optimize.minimize#scipy.optimize.minimize) heesp has the signature of:

> hessp(x, p, *args) ->  ndarray shape (n,)

which is what is expected by `optimize.minimize`.